### PR TITLE
added test with new cross platform app 

### PR DESCRIPTION
**Pull request by saikrishna321**
@jlipps as discussed i have refactored AndroidDragAndDrop with the new app(iOS and android)

Execution Command:
mvn clean -Dtest=AndroidDragAndDrop test

Application repo:

https://github.com/saikrishna321/VodQAReactNative

### CODE REVIEW

**Comment by jlipps** [comment]
@saikrishna321 you'll need to close this and resubmit with the commit coming from your github user's e-mail address (see instructions from clabot). meanwhile i'll give a review

File: sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.junit.Before;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Created by saikrisv on 01/06/17.
```
Comment (jlipps) [inline sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java]: as a rule we don't include personal details like this in submitted code

File: sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ */
+public class BaseCrossDriver {
+    public AppiumDriver driver;
+    private static AppiumDriverLocalService service;
+
+    @Before
+    public void setUp() throws Exception {
+        service = AppiumDriverLocalService.buildDefaultService();
```
Comment (jlipps) [inline sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java]: do we really want the sample code to be starting appium for us? most of the languages have appium started separately. but if this is the way people tend to do it in JAVA, I'm ok with that. we should then have a readme explaining how to run it

File: sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if (service == null || !service.isRunning()) {
+            throw new AppiumServerHasNotBeenStartedLocallyException(
+                    "An appium server node is not started!");
+        }
+        iOSCaps();
+    }
+
+    private void androidCaps() throws IOException {
+        File classpathRoot = new File(System.getProperty("user.dir"));
+        File appDir = new File(classpathRoot, "../../../apps/");
```
Comment (jlipps) [inline sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java]: not sure how java path resolution works; will this be OK on Windows?

File: sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java (hunk 1 above)
Comment (saikrishna321) [inline sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java]: my bad will remove this in the new PR

File: sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java (hunk 2 above)
Comment (saikrishna321) [inline sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java]: Yes, as java-client has the feasibility we tend to do this.

File: sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java (hunk 3 above)
Comment (saikrishna321) [inline sample-code/examples/java/junit/src/test/java/com/saucelabs/appium/BaseCrossDriver.java]: Have tested on windows, works fine

**Pull request closed without merging**